rct229/rulesets/ashrae9012019/section12/section12rule3.py

```diff
@@ -11,46 +33,204 @@ class Section12Rule3(RuleDefinitionListIndexedBase):
+                        required_fields={
+                            "$": [
+                                "has_automatic_control",
+                                "multiplier_schedule",
+                            ]
+                        },
+                        manual_check_required_msg="Credit for automatic receptacle controls was expected, but baseline and proposed miscellaneous equipment schedules are identical.",
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    misc_equip_b = context.BASELINE_0
+                    misc_equip_p = context.PROPOSED
+
+                    is_leap_year = data["is_leap_year"]
+                    space_type_p = data["space_type_p"]
+                    schedule_b = data["schedule_b"]
+                    schedule_p = data["schedule_p"]
+
+                    if (
+                        space_type_p not in EXPECTED_RECEPTACLE_CONTROL_SPACE_TYPES
+                        and misc_equip_p["has_automatic_control"]
+                    ):
+                        expected_receptacle_power_credit = 0.1 * getattr_(
+                            misc_equip_p,
+                            "miscellaneous_equipment",
+                            "automatic_controlled_percentage",
+                        )
+
+                        hourly_multiplier_schedule_b = misc_equip_b[
+                            "multiplier_schedule"
+                        ]
+                        hourly_multiplier_schedule_p = misc_equip_p[
+                            "multiplier_schedule"
+                        ]
+
+                        expected_hourly_values = [
+                            hour_value * (1 - expected_receptacle_power_credit)
+                            for hour_value in schedule_b[hourly_multiplier_schedule_b]
+                        ]
+
+                        mask_schedule = (
+                            [1] * LeapYear.LEAP_YEAR_HOURS
+                            if is_leap_year
+                            else [1] * LeapYear.REGULAR_YEAR_HOURS
+                        )
+
+                        credit_comparison_data = compare_schedules(
+                            expected_hourly_values,
+                            schedule_p[hourly_multiplier_schedule_p],
+                            mask_schedule,
+                            is_leap_year,
+                        )["total_hours_matched"]
+
+                        no_credit_comparison_data = compare_schedules(
+                            schedule_b[hourly_multiplier_schedule_b],
+                            schedule_p[hourly_multiplier_schedule_p],
+                            mask_schedule,
+                            is_leap_year,
+                        )["total_hours_matched"]
+
+                    return {
+                        "expected_hourly_values_len": len(expected_hourly_values),
+                        "credit_comparison_data": credit_comparison_data,
```

Comment:
this is confusing - use `credit_comparison_total_hours_matched` and `no_credit_comparison_total_hours_matched`

Reply:
Addressed.